base/Base.jl

```diff
@@ -25,6 +25,11 @@ function include(mod::Module, path::String)
 end
 include(path::String) = include(Base, path)
 
+struct IncludeInto
```

Comment:
These should be subtypes of Function. Also, if we make them use the function keyword syntax (so a closure), they will get their own MethodTable, which may be slightly better?

Reply:
I don't think the function keyword should be required. Just the supertype is sufficient to get special method table:
https://github.com/JuliaLang/julia/blob/dd310849adbf9f089d7e21c142b513deb8ff7b01/src/datatype.c#L846

Reply:
super is null there unless you use the keyword

Reply:
Oh, it's that case where we want to have the binding available while we evaluate the supertype ... Sigh. I don't know if splitting the mt is really worth the additional definition complexity. These are not called that much.